File: sorts/insertion sort/InsertionSort.cpp
#include<iostream>
using namespace std; 
int main()
{
	int FirstUnsortedIndex, i, j,  array[100], size,k, newElement; 
	cout<<"Enter the size of the array"<<endl;
	cin>>size;
	cout<<"Enter the elements into the array"<<endl;
	for(k = 0; k<size;k++)
	{
		cin>>array[k];
	}
	//For the insertion sort logic, refer to notes
	for(FirstUnsortedIndex = 1; FirstUnsortedIndex<size - 1; FirstUnsortedIndex++)
	{
		newElement = array[FirstUnsortedIndex];
		for(i = FirstUnsortedIndex; i > 0 && array[i - 1] > newElement; i--)
		{
			array[i] = array[i - 1];
		}
		array[i] = newElement;
	}
	cout<<endl;
	for(j = 0; j < size; j++)
	{
		cout<<array[j]<<endl;
	}
}

File: sorts/insertion sort/InsertionSort.java
import java.util.Scanner;
class InsertionSort
{
	public static void main(String[] args)
	{
		//Collection of data
			/*	To collect the data as an input from the user, remove these comments
				
				int array[] = new int[10];
				Scanner in = new Scanner(System.in);
				System.out.println(Ënter the size of array:");
				int n = in.nextInt();
				System.out.println(" Enter the elements:");
				for(int i=0;i<n;i++)
				{
					array[i] = in.nextInt();
				}
				
			*/
			
			// Hard coding
			int array[] = {20,35,-15,7,55,-1,-22};
		// For insertion sort theory refer to Insertion Sort notes
		for(int FirstUnsortedIndex = 1; FirstUnsortedIndex < array.length ; FirstUnsortedIndex++)
		{
			int newElement = array[FirstUnsortedIndex];
			int i;
			for(i = FirstUnsortedIndex; i > 0 && array[i - 1] > newElement; i--)
			{
				array[i] = array[i - 1];
			}
			array[i] = newElement;
		}
		for(int element: array)
		{
			System.out.println(element);
		}
	}
}